src/internal/event.ts

```diff
@@ -10,41 +10,80 @@
 
 import { AptosConfig } from "../api/aptos_config";
 import { AccountAddress } from "../core";
-import { AnyNumber, GetEventsResponse, HexInput } from "../types";
+import { AnyNumber, GetEventsResponse, HexInput, IndexerPaginationArgs, MoveResourceType, OrderBy } from "../types";
 import { GetEventsQuery } from "../types/generated/operations";
 import { GetEvents } from "../types/generated/queries";
+import { EventsBoolExp } from "../types/generated/types";
 import { queryIndexer } from "./general";
 
-/**
- * Get events by creation number and the address
- *
- * @param args.aptosConfig - The aptos config
- * @param args.address - The account address
- * @param args.creationNumber - The event creation number
- * @returns Promise<GetEventsByCreationNumberResponse>
- */
-export async function getEventsByCreationNumber(args: {
+export async function getAccountEventsByCreationNumber(args: {
   aptosConfig: AptosConfig;
   address: HexInput;
   creationNumber: AnyNumber;
 }): Promise<GetEventsResponse> {
   const { aptosConfig, creationNumber } = args;
   const address = AccountAddress.fromHexInput({ input: args.address }).toString();
 
-  const whereCondition: any = {
+  const whereCondition: EventsBoolExp = {
     account_address: { _eq: address },
     creation_number: { _eq: creationNumber },
   };
 
+  const data = await getEvents({ aptosConfig, options: { where: whereCondition } });
+
+  return data;
+}
+
+export async function getAccountEventsByEventType(args: {
+  aptosConfig: AptosConfig;
+  address: HexInput;
+  eventType: MoveResourceType;
+  options?: {
+    pagination?: IndexerPaginationArgs;
+    orderBy?: OrderBy<GetEventsResponse[0]>;
+  };
+}): Promise<GetEventsResponse> {
+  const { aptosConfig, eventType, options } = args;
+  const address = AccountAddress.fromHexInput({ input: args.address }).toString();
+
+  const whereCondition: EventsBoolExp = {
+    account_address: { _eq: address },
+    type: { _eq: eventType },
+  };
+
+  const customOptions = {
+    where: whereCondition,
+    pagination: options?.pagination,
+    orderBy: options?.orderBy,
+  };
+
+  const data = await getEvents({ aptosConfig, options: customOptions });
+
+  return data;
```

Comment:
Can drop the await and return, and just return the future directly

```suggestion
  return getEvents({ aptosConfig, options: customOptions });
```